dbos/_queue.py

```diff
@@ -105,10 +113,21 @@ def queue_thread(stop_event: threading.Event, dbos: "DBOS") -> None:
         queues = dict(dbos._registry.queue_info_map)
         for _, queue in queues.items():
             try:
-                wf_ids = dbos._sys_db.start_queued_workflows(
-                    queue, GlobalParams.executor_id, GlobalParams.app_version
-                )
-                for id in wf_ids:
+                if queue.partition_queue:
+                    dequeued_workflows = []
+                    queue_partition_keys = dbos._sys_db.get_queue_partitions(queue.name)
```

Comment:
I'm mostly worried about the overhead of this query. Instead of doing it per queue, can we just have a single roundtrip to retrieve partitions for all queues?

Reply:
This creates a worse `SELECT DISTINCT`, so not likely to be an optimization